### PR TITLE
[psqldef] Recognize schema for types

### PR DESCRIPTION
**Pull request by wreulicke**
[psqldef] Recognize schema for types.

This fixes the bug that generates wrong alter statements for types when there are types of same name in different schema.

also see: regression test in first commit
https://github.com/sqldef/sqldef/actions/runs/9403732908/job/25900968987

### CODE REVIEW

**Comment by wreulicke** [comment]
Tests with Postgres 10 and 11 failed. 👀
It seems PostgreSQL 12 or above supports `alter type ... in transaction`.
https://stackoverflow.com/a/53150621

File: cmd/testutils/testutils.go [hunk 1]
```diff
@@ -70,6 +71,13 @@ func RunTest(t *testing.T, db database.Database, test TestCase, mode schema.Gene
 		t.Skipf("Version '%s' is larger than max_version '%s'", version, test.MaxVersion)
 	}
 
+	if test.Init != "" {
+		_, err := db.DB().Exec(test.Init)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
```
Comment (wreulicke) [inline cmd/testutils/testutils.go:79]: Added to prepare schema

**Comment by wreulicke** [comment]
@k0kubun 
Hello, thank you for maintaining a great tool, sqldef. 
I faced issues with types in Postgres.

Current implementation doesn't care about types defined in different schemata.
So, sqldef generates wrong alter statements when the same name types are in different schemata.
This PR fixes the issue.

Can you check this PR?

File: database/postgres/database.go [hunk 2]
```diff
@@ -233,17 +234,20 @@ func (d *PostgresDatabase) types() ([]string, error) {
 
 	var ddls []string
 	for rows.Next() {
-		var typeName, labels string
-		if err := rows.Scan(&typeName, &labels); err != nil {
+		var typeSchema, typeName, labels string
+		if err := rows.Scan(&typeSchema, &typeName, &labels); err != nil {
 			return nil, err
 		}
+		if d.config.TargetSchema != "" && d.config.TargetSchema != typeSchema {
```
Comment (hokaccha) [inline database/postgres/database.go:241]: The type of TargetSchema has changed due to the modifications in https://github.com/sqldef/sqldef/pull/540. Please merge the master branch to address this change 🙏

File: cmd/psqldef/tests.yml [hunk 3]
```diff
@@ -162,6 +162,32 @@ AlterTypeAddValue:
     ALTER TYPE public.eventtype ADD VALUE 'send_meeting_options';
     ALTER TYPE public.eventtype ADD VALUE 'update_reactionslot';
   min_version: '12'
+AlterTypeAddValueWithSameTypeNameInDifferentSchema:
+  init: |
+    CREATE SCHEMA schema1;
+    CREATE SCHEMA schema2;
```
Comment (hokaccha) [inline cmd/psqldef/tests.yml:168]: The `init` is no longer needed as the `CREATE SCHEMA` is supported by the changes in https://github.com/sqldef/sqldef/pull/541. Put it directly in `current`.

**Review by hokaccha — COMMENTED** [review]
LGTM, but please make a few corrections.

**Comment by wreulicke** [comment]
@hokaccha 
Sorry for my late reply.
I will try to fix those soon.

File: cmd/psqldef/tests.yml (hunk 3 above)
Comment (wreulicke) [reply cmd/psqldef/tests.yml:168]: @hokaccha 
I fixed this at [a7975f3](https://github.com/sqldef/sqldef/pull/532/commits/a7975f3e997424837fb1445cadb2d69431e36f91) and [8cca999](https://github.com/sqldef/sqldef/pull/532/commits/8cca99938a0d680e48ac47a33acb64cf641a236b).

**Comment by wreulicke** [comment]
@hokaccha 
I fixed. Could you check this PR again?

**Review by hokaccha — APPROVED** [review]
lgtm, thanks!

**Pull request merged**